ui.go

```diff
@@ -804,6 +804,24 @@ func (ui *ui) drawPromptLine(nav *nav) {
 	ui.promptWin.print(ui.screen, 0, 0, st, prompt)
 }
 
+func addStatLineVar(ruler []string, name string) []string {
+	if !strings.HasPrefix(name, "lf_") {
+		return ruler
+	}
+
+	val := os.Getenv(name)
```

Comment:
Why not get the value from the `gOpts` stuct? I'm sure you considered this question already, but it wasn't immediately clear to me.

Reply:
I considered this too, but the `gOpts` struct contains values that aren't strings, and I found it easier to just use the enviornment variables rather duplicate the logic in `exportOpts`. I do agree there is a possible performance concern with this though, I'll spend some more time to see get this working without adding too much complexity.

Reply:
Sounds reasonable.

I'm more worried about correctness than performance: is it guaranteed on every system that `os.Gentenv` will see the values exported by the `exportOpts`? I was suspecting that `os.Setenv` only affects child processes. That suspicion was probably wrong as this code seems to work on your system, at least (I haven't played with it yet).

It now seems likely to me that this *is* guaranteed on every OS Go supports (@gokcehan might know or it could be in the docs), and that this code is fine. I am still a little worried -- if this doesn't always hold, it could lead to subtle bugs.


--------------

Here's another approach, very possibly what you're already trying: create a function `optsStringMap()` that converts `gOpts` into a map with string values, and then have both your code and `exportOpts` call that function.

Reply:
> Here's another approach, very possibly what you're already trying: extract most of the exportOpts code into a new function optsStringMap(). That function would convert gOpts into a map with string values, and then have both your code and (now much shorter) exportOpts call that function.

I originally didn't want to touch any of the `exportOpts` code, but I gave in and ended up implementing this in 9f8f21f. Thanks for the suggestion.